src/web/js/ide.js

```diff
@@ -31,32 +48,19 @@ const runtimeApi = {
     // TODO(joe): pass in a URL to uniquely identify this program
     if(!url) { url = "definitions://"; }
     var runtime = window.CPOIDEHooks.runtime;
-    function getVal(mod, name) {
-      return runtime.getField(runtime.getField(mod, "values"), name);
-    }
-    var parse = getVal(window.CPOIDEHooks.parsePyret, "surface-parse");
+    // TODO(joe): is "this" trustworthy here?
+    var self = this;
+    var parse = this._getVal("parsePyret", "surface-parse");
     return new Promise((resolve, reject) => {
-      function handleError(parseResult) {
-        runtime.runThunk(
-          () => runtime.toRepr(parseResult.exn.exn),
-          (exnResult) => {
-            if(runtime.isSuccessResult(exnResult)) {
-              reject(exnResult.result);
-            }
-            else {
-              console.error("Could not render: ", parseResult, " because ", exnResult);
-              reject("An error occurred while rendering a parse error, details logged to console");
-            }
-          });
-      }
       runtime.runThunk(
         () => parse.app(src, url),
         (parseResult) => {
           if(runtime.isSuccessResult(parseResult)) {
             resolve(parseResult.result);
           }
           else {
-            handleError(parseResult);
+            // NOTE(joe): intentionally passing reject twice; want to report an error either way
+            self._toReprOrDie(runtime, parseResult.exn.exn, reject, reject);
```

Comment:
Yeah Ben pointed out my stupid.  It should call resolve in the success case and reject in the failure case, but when rendering an error just get two rejects.  I'll fix.